crates/forge/tests/integration/trace.rs

```diff
@@ -0,0 +1,757 @@
+                    },
+                ];
+            
+                assert(trace == expected_trace, 'proxy deploy');
+            }
+        "#
+        ),
+        Contract::from_code_path(
+            "TraceInfoProxy".to_string(),
+            Path::new("tests/data/contracts/trace_info_proxy.cairo"),
+        )
+        .unwrap(),
+        Contract::from_code_path(
+            "TraceInfoChecker".to_string(),
+            Path::new("tests/data/contracts/trace_info_checker.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+#[allow(clippy::too_many_lines)]
+fn trace_call() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use snforge_std::{declare, ContractClassTrait, test_address};
+            use snforge_std::trace::{CallEntryPoint, CallType, EntryPointType, get_last_call_trace};
+            
+            use starknet::{ContractAddress, ClassHash};
+            
+            #[starknet::interface]
+            trait ITraceInfoProxy<T> {
+                fn with_libcall(self: @T, class_hash: ClassHash) -> felt252;
+                fn regular_call(self: @T, contract_address: ContractAddress) -> felt252;
+                fn with_panic(self: @T, contract_address: ContractAddress);
+                fn call_two(self: @T, checker_address: ContractAddress, dummy_address: ContractAddress);
+            }
+            
+            #[starknet::interface]
+            trait ITraceInfoChecker<T> {
+                fn from_proxy(self: @T, data: felt252) -> felt252;
+                fn panic(self: @T);
+            }
+            
+            #[starknet::interface]
+            trait ITraceDummy<T> {
+                fn from_proxy(ref self: T);
+            }
+            
+            #[test]
+            fn test_call_trace_info() {
+                let proxy = declare('TraceInfoProxy');
+                let checker = declare('TraceInfoChecker');
+                let dummy = declare('TraceDummy');
+            
+                let checker_address = checker.deploy(@array![]).unwrap();
+                let proxy_address = proxy.deploy(@array![checker_address.into()]).unwrap();
+                let dummy_address = dummy.deploy(@array![]).unwrap();
+            
+                let proxy_dispatcher = ITraceInfoProxyDispatcher { contract_address: proxy_address };
+            
+                proxy_dispatcher.regular_call(checker_address);
+                assert_trace_after_proxy_regular_call(get_last_call_trace(), proxy_address, checker_address);
+            
+                proxy_dispatcher.with_libcall(checker.class_hash);
+                assert_trace_after_proxy_call_with_libcall(
+                    get_last_call_trace(), proxy_address, checker.class_hash
+                );
+            
+                proxy_dispatcher.call_two(checker_address, dummy_address);
+                assert_trace_after_proxy_call_two(
```

Comment:
> Are scenarios like tests calls A, A calls B and C tested?

@MaksymilianDemitraszek here is this exact scenario, in `trace_library_call_from_test` too